docs/Menu-Commands.md

```diff
@@ -0,0 +1,78 @@
+  1. in `package.nls.json`, define the user-facing string for the key. (For example: `"uss.addFavorite": "Add to Favorites"`.)
+
+  - More details on localization can be found in the [Developer's ReadMe](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/Developer's%20ReadMe.md#adding-strings).
+
+- `category` property: The value of this should be "Zowe Explorer".
+
+**Note:** By default, defined commands will appear in VS Code's Command Palette. If the command being added is not designed to work from the Command Palette, be sure to add an entry for the command in the `contributes.menus.commandPalette` section of `packages/zowe-explorer/package.json`, and specify `"when": "never"`. (For more details, see VS Code's documentation on [specifying command visibility in the Command Palette](https://code.visualstudio.com/api/references/contribution-points#Context-specific-visibility-of-Command-Palette-menu-items).)
+
+## Contributing context menu items
+
+Context menus appear when the user right-clicks on an item (for example, a data set or profile). Commands can be added into the context menus by creating an entry for a [defined command](#defining-a-new-command) in the `menus contributes.menus.view/item/context` section of `packages/zowe-explorer/package.json`.
+
+- If adding a command to a Zowe Explorer context menu, be sure to assign a value to the `group` property of the entry. This helps provide visual separation between different command categories in the UI. You can see examples of context menu groups in the current menu items in Zowe Explorer's `package.json`. Follow the steps below when assigning an entry's `group` property:
+
+  1. First, check the `package.json` to see if there is an existing context menu group that the new command can intuitively be categorized under (from an end-user's perspective).
+  1. If the new command does not fit under any existing context menu groups, create a new context menu group by assigning the `group` property a new value that follows [Zowe Explorer's context menu group naming convention](#zowe-explorer-context-menu-group-naming-convention).
+
+For general technical details on grouping context menu items, see the following VS Code documentation:
+
+- [Sorting context menu groups](https://code.visualstudio.com/api/references/contribution-points#Sorting-of-groups)
+- [Sorting commands within a context menu group](https://code.visualstudio.com/api/references/contribution-points#Sorting-inside-groups)
+
+### Zowe Explorer context menu group naming convention
+
+The prefix `##_zowe_` is reserved for use with core Zowe Explorer command groups. Group names for Zowe Explorer are formatted as follows: `##_zowe_<view><groupSpecifier>`. For example, commands related to creating items in Zowe Explorer's Data Sets view are in the group named `01_zowe_mvsCreate`.
+
+A breakdown of this naming convention is described below.
```

Comment:
The first 2 points in this section are confusing to me. When I'm adding an extender command to the menu, should I use `##_zowe_` or not? It says this is "reserved for core commands", but it also says group names are formatted using that prefix.

I would maybe add a note telling people _not_ to use ## and to _definitely_ use `zowe_<view><groupSpecifier>` in their own commands. Why these rules exist are not important to extenders, imho. They will probably figure out the alphabetical ordering of commands, and the fact that Zowe Explorer commands always show up first in the list, on their own. And if they do not use this format then they simply won't be conformant.

In regards to the documentation below that, listing the views and group specifiers, are these mandatory options or not? I mean, do I have to use one of these views and one of these specifiers, or can I come up with my own?

I assume the view must be one from the list, but the specifier can be something outside of this list? Could we make sure to specify which portions of this format are mandatory and which are not? And if the specifier is _not_ mandatory, maybe we don't need to list so many examples. People can just look at the `package.json` file for more examples.

Reply:
@katelynienaber This documentation is guidance for Zowe Explorer contributors, not for extenders. For extenders, their guidance is in the conformance criteria (PR #1290), where we tell them to avoid using the `##_zowe_` prefix in their command group names.

The section about views and group specifiers is written based on Zowe Explorer's current command group names. I think if a new Zowe Explorer command can fit into one of the current command groups, the developer should put it there rather than creating a whole new group. The goal of documenting the currently-used views and group specifiers is to make it easier and quicker for a Zowe Explorer developer to know what groups are available to pick from (or if they maybe need to create a new group). However, if you think it is confusing, perhaps we can consider a different approach to this.